microprofile/WFLY-15832-microconfig-config-root-folder.adoc

```diff
@@ -0,0 +1,99 @@
+= WFLY-15832 - MP Config: Ability to specify root config source directory
+:author:            Kabir Khan
+:email:             kkhan@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+The MicroProfile Config with SmallRye subsystem currently supports configuration of several types of Config Sources. One of these allows you to specify a directory, wherein each file is used as a config property. The file name is the name of the property, and the contents of the file is the value of the property.
+
+This is useful on OpenShift where ConfigMaps, the ServiceBinding opertator (and probably other things) expose config via this mechanism for use in applications.
+
+This RFE is about enhancing the subsystem to allow specifying a root directory, of which the top-level directories under it will appear as Comfig Sources. The motivation is that each 'thing' on OpenShift installs their config under a root directory. e.g. in the ServiceBinding operator case this is under the `/bindings` directory and ConfigMaps are installed under `/etc/config`.
+
+So if you have two ConfigMaps called `cfg-a` and `cfg-b`, their respective directories containing the config files will be `/etc/config/cfg-a` and `/etc/config/cfg-b`. At present this means we need to add two Config Source entries to the subsystem, and we need to know the exact names of both, in order to pick these up.
+
+While we will keep the existing mechanism, the proposal is to be able to specify a directory as a root for configuration, and that each top level directory becomes a Config Source as with the previous mechanism.
+
+This has the advantage of only needing to specify one Config Source root in order to pick up all ConfigMaps, one to pick up all ServiceBinding configs, and so on. An added bonus is not needing to know the exact names of the configuration directories, which are in some cases autogenerated.
```

Comment:
Are we going to support 1 level of recursion, right?

Reply:
Ok, answer to my  question is here https://github.com/wildfly/wildfly-proposals/pull/450/files#diff-9f1354c78479d2107750d078d4f846420ac82216e6eda99d34a0ad36ea2d948fR74. Only one level of recursion. +1